resources/static/js/account.js

```diff
@@ -86,72 +94,90 @@ function showRamp() {
 }
 
 async function checkWeb3Network() {
-  let required_chain_id = CHAIN_ID;
-  await connectWeb3();
-  web3.version.getNetwork(function (error, chain_id) {
-    if (error) {
-      console.error(error);
-    }
-
-    if (chain_id != required_chain_id) {
-      let current_chain_name = CHAIN_ID_MAPPING[chain_id];
-      let required_chain_name = CHAIN_ID_MAPPING[required_chain_id];
-      alert(
-        `Web3 Browser connected to ${current_chain_name}, please change to ${required_chain_name}.`
-      );
-    }
-  });
+  const requiredChainID = CHAIN_ID;
+  const currentChainID = parseInt(await provider.request({ method: 'eth_chainId' }), 16);
+  if (currentChainID != requiredChainID) {
+    const currentChainName = CHAIN_ID_MAPPING[currentChainID];
+    const requiredChainName = CHAIN_ID_MAPPING[requiredChainID];
+    alert(
+      `Web3 Browser connected to ${currentChainName}, please change to ${requiredChainName}.`
+    );
+    return false;
+  }
+  return true;
 }
 
-function makeWeb3Transaction(w3, transaction_data) {
-  w3.eth.sendTransaction(transaction_data, function (error, result) {
-    if (result) {
-      // result is the transaction hash
-      trackTransaction(result, CONFIGURATION_FILE_NAME);
+async function connectWeb3() {
+  let accounts = [];
+  try {
+    accounts = await provider.request({ method: 'eth_accounts' });
```

Comment:
Should this even be called before calling `eth_requestAccounts`, or will this fail all the time anyways?
I read it that way that `eth_requestAccounts` **MAY eventually** trigger the UI-element - meaning that this is only the case for the first access-request? I could be wrong - so I'm mainly just curious.

Reply:
The user may already have given the allowance to access the accounts, when the button is clicked. For example in case the button is clicked a second time after canceling the first transaction. 
In my understanding, every call to `eth_requestAccounts` may trigger a user interface, depending on the RPC Provider. Since this is not the case with MetaMask, I think we can just use the call to `eth_requestAccounts`.

Reply:
Ok - I couldn't find anything about `eth_accounts` being unlocked by the `eth_requestAccounts` as well, 
and the example code from the metamask docs was using `eth_requestAccounts` just before a transaction.